db/migrate/20190201222556_remove_facility_id_from_accounts.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class RemoveFacilityIdFromAccounts < ActiveRecord::Migration[5.0]
+  def up
+    remove_foreign_key :accounts, :facilities
+    remove_column :accounts, :facility_id
+  end
+
+  def down
+    add_reference :accounts,
+                  :facility,
+                  after: "remittance_information",
+                  foreign_key: { name: "fk_account_facility_id" },
+                  index: { name: "fk_account_facility_id" }
+
+    # If there happen to be more than one join, then :shrug:, only one will get set.
+    execute <<~SQL
+      UPDATE accounts
+      SET accounts.facility_id = (
+        SELECT facility_id
+        FROM account_facility_joins
+        WHERE accounts.id = account_facility_joins.account_id
+        LIMIT 1
```

Comment:
Since we know that the oldest row in `account_facility_joins` came from the migration we did to populate that table, can we add an `ORDER BY created_at DESC` here, and we’ll be certain that we’re restoring the data correctly? That way, we don’t need the 🤷‍♂️ comment, and we’re not restoring a random associated facility but the one where the account was created originally.

Reply:
Except where they removed the original facility and then re-added it. And we're still technically losing data, so I think a comment is probably still warranted. But yeah, the `ORDER BY` sounds like a good plan to cover everything but a super weird case (and we're hopefully not going to need the `down` anyways).